File: compute_duration.c
#include<stdio.h>


int main(int argc, char* argv[]){
    
    printf("============================================================\n\n");
    printf("     Compute Durations:\n\n");
	printf("This program was written by Stephan Lenz and may only be\n");
	printf("used for educational purposes.\n\n");
	printf("============================================================\n\n");
	
	if(argc < 2){
		printf("Error: No filename given\n");
		return 1; 
	}


	FILE* file;
	file = fopen(argv[1], "r");
	
	if(file == NULL){
		printf("Error: File not existent!\n");
		return 1;
	}
	
	char line[80];
	int start = 0;
	int end   = 0;
	int duration = 0;
	
	while(fgets(line, 80, file) != NULL)
	{
		/* get a line, up to 80 chars from fr.  done if NULL */
		sscanf (line, "%d %d", &start, &end);
        duration = end - start;
		printf ("%d years\n", duration);
	}

	fclose(file);


	return 0;
}

File: compute_duration.java
import java.util.Scanner;
import java.io.File;

public class compute_duration {
    public static void main(String [] args) throws Exception {
        
        System.out.println("===================================================");
        System.out.println("     Compute Durations:");
	    System.out.println("This program was written by Stephan Lenz and may");
	    System.out.println("only be used for educational purposes.");
	    System.out.println("===================================================");
        
	    Scanner fileScanner = new Scanner( new File(args[0]) );
		
        int start = 0;
	    int end   = 0;
	    int duration = 0;
	
	    while(fileScanner.hasNextInt())
	    {
		    start = fileScanner.nextInt();
		    end = fileScanner.nextInt();
            duration = end - start;
		    System.out.println(Integer.toString(duration) + " years");
	    }
    }
}
